### PR TITLE
3.0 Release

### PR DESCRIPTION
**Pull request by merbla**
* #91 & #92 - Fix cherrypick from master
 * #97 - Support of ` IConfiguration` parameters & `IConfigurationSection` parameters
 * #83 - Updated dependencies of `Microsoft.Extensions.DependencyModel`, `Microsoft.Extensions.Configuration.Abstraction` & `Microsoft.Extensions.Options.ConfigurationExtensions` per TFM
 * Fix for #98 - specify string array params
 * Target Framework change to `netcoreapp2.0`
 * Build updates including addition of Travis Build
 * #105 - detect and fail on ambiguous configurations
 * #110 - destructure support
 * #111 - case-insensitive argument matching
 * #132 - choose string overloads to resolve binding ambiguities
 * #134 - specify repository URL in package
 * #124 - build a .NET 4.6.1 target
 * #136 - control assembly source
 * #138 - remove unnecessary package ref
 * #139 - remove unused class
 * #140 - expand support for destructure/enrich/filter configuration

### CODE REVIEW

**Comment by tsimbalar** [comment]
It's green now, I can finally go on with my life :)

File: src/Serilog.Settings.Configuration/Serilog.Settings.Configuration.csproj [hunk 1]
```diff
@@ -16,20 +16,22 @@
...
-  </PropertyGroup>
+  <ItemGroup Condition="'$(TargetFramework)' == 'net451'">
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="1.1.2" />
+    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="1.1.2" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
+    <PackageReference Include="Microsoft.Extensions.Configuration.Abstractions" Version="2.0.1" />
+    <PackageReference Include="Microsoft.Extensions.Options.ConfigurationExtensions" Version="2.0.0" />
+  </ItemGroup>
```
Comment (skomis-mm) [inline src/Serilog.Settings.Configuration/Serilog.Settings.Configuration.csproj]: The `Microsoft.Extensions.Options.ConfigurationExtensions` dependency includes `.Abstrations` already along with the:
- Microsoft.Extensions.Configuration
- Microsoft.Extensions.Configuration.Binder,
- Microsoft.Extensions.DependencyInjection.Abstractions
- Microsoft.Extensions.Options

It is good for [this](https://github.com/serilog/serilog-settings-configuration/pull/112/files#diff-6940a2e06a5cf00461ee3a198b259c07R54) one, but I wonder if this all was considered. If yes, then `.Abstractions` dependency could be omitted.

**Comment by merbla** [comment]
@tsimbalar @skomis-mm @nblumhardt this one has been open since May, thoughts on when we should release?

**Comment by nblumhardt** [comment]
I think once @MV10 's #63 is in, we should hit the button. Would definitely benefit from a close look over for anything we've collectively missed, though.

**Comment by tsimbalar** [comment]
Yep, let's ship it when possible, there will be plenty of time for minor releases later on :)

**Comment by nblumhardt** [comment]
Last PR is in. Just need to compile list of changes, and eyeball it all for one final sanity-check :-)

**Comment by nblumhardt** [comment]
Ah and also this item from @skomis-mm - https://github.com/serilog/serilog-settings-configuration/pull/112#discussion_r203360284 - anyone keen to dig in and see how much of that dependency list can be whittled away?

**Comment by MV10** [comment]
@skomis-mm was correct, merging a PR to remove that refrerence. SHIP IT!

**Comment by tsimbalar** [comment]
Finally found the time to do #139 and #140 . Fill free to include them in v3.0 as well, but they can also wait for the next minor releaser ;)

**Comment by nblumhardt** [comment]
LGTM :-)

**Comment by nblumhardt** [comment]
Going to hit the button on this, and keep a close eye on the repo over the next few days. 👏 to everyone who contributed to this - it's a pretty substantial update!

Great that the wheels are turning again; now that it's close to feature-complete, I think there's still a lot of opportunities to improve ergonomics of the JSON syntax - #49 is desperate for some attention ;-)

**Comment by MV10** [comment]
I've been thinking about messing around with a #49 implementation (specifically my preference for the "brackety" no-hidden-syntax approach). I have a couple more things I want to wrap up in the SQL sink but I think later this week I can take a stab at it.

**Pull request merged**